@here/harp-mapview/lib/text/TextElementsRenderer.ts

```diff
@@ -122,6 +122,13 @@ const OVERLOAD_UPDATE_TIME_LIMIT = 5;
  */
 const OVERLOAD_PLACE_TIME_LIMIT = 10;
 
+// Allowed number of pixels outside the screen to test for placement.
+// A value of 0 will lead to labels popping in at the border of the screen. A large value will
+// lead to many labels being placed outside the screen, with all the required actions for measuring
+// and loading glyphs.
+const MAX_HORIZONTAL_DISTANCE_TO_BORDER = 100;
+const MAX_VERTICAL_DISTANCE_TO_BORDER = 40;
```

Comment:
It's very difficult these absolute thresholds are gonna fit all cases: different fonts, font sizes, label lengths, clients...
As in harp native, I would put this as a styling property, so that it can be adjusted on a case by case basis.
I would also use thresholds that are relative to the total label size. Harp native has a "minVisibility" factor [0,1], representing how much area of the label must be inside the screen before it's made visible.

Reply:
> The alternative is to do layout for **all** labels on a tile, no? 
> Ass described below, this performance optimization has been removed against my judgement.

It's not the only alternative, I'm not sure why you think I asked you to do that. Quoting my comment, this was my suggestion:

> As in harp native, I would put this as a styling property, so that it can be adjusted on a case by case basis.
> I would also use thresholds that are relative to the total label size. Harp native has a "minVisibility" factor [0,1], representing how much area of the label must be inside the screen before it's made visible.

So I'm just suggesting to change label bbox culling to use **relative** thresholds instead of the absolute ones you were using.
I also mentioned in my last comment where you can find HARP native's code to cull labels:

> Please have a look at how harp native checks the label bounds visibility:
> https://gerrit.it.here.com/gitweb?p=harp/core.git;a=blob;f=label/src/placement/Positioning.cpp;h=32481bd86c16f78ceac048b50895e29a9c181799;hb=HEAD

Reply:
I was thinking that, to do a proper computation of the label area the layout has to be computed before the visible part can be computed. The measuring of text is pretty expensive due to the complex machinery required, with fonts and glyphs required to be loaded in addition to placement checks to be done for optional layout positions. If you have an approach that does not require that, I am open for that.
I would rather go for something that is fast and works fine for the standard settings, and live with minor visual issues in border cases with unusual large fonts or such things.

Reply:
This code has been removed. I suggest that we measure the impact of this PR, and optimize it in another PR if necessary.

Reply:
> I was thinking that, to do a proper computation of the label area the layout has to be computed before the visible part can be computed. The measuring of text is pretty expensive due to the complex machinery required, with fonts and glyphs required to be loaded in addition to placement checks to be done for optional layout positions. If you have an approach that does not require that, I am open for that.
> I would rather go for something that is fast and works fine for the standard settings, and live with minor visual issues in border cases with unusual large fonts or such things.

That's clear, as I said, disregard my first comment, it's really unrelated to what you were trying to achieve here. I agree we should early reject labels that are far away from the screen.

> This code has been removed. I suggest that we measure the impact of this PR, and optimize it in another PR if necessary.

Ok, makes sense to reduce the scope of this PR. But we can only do it if FPS is not impacted. Please, run manual perf tests on macbook to verify there's no noticeable drop in performance:
https://render-verity.jenkins.release.in.here.com/view/Tests/job/verity-perf-device-test-macbook-manual

Reply:
I ran a test, and it showed some impact. I would strongly suggest to put the optimization into another PR.